### PR TITLE
Using updated wallet & miner APIs

### PR DESCRIPTION
**Pull request by DavidBurkett**

### CODE REVIEW

File: src/wallet/walletdb.cpp [hunk 1]
```diff
@@ -800,19 +800,31 @@ struct MWCoin {
...
+                READWRITE(coin.key->keyBytes);
+            }
+        }
     }
 };
 
 bool WalletBatch::WriteMWCoin(const libmw::Coin& coin)
 {
-    return WriteIC(std::make_pair(std::string("mweb_tx"), coin.commitment), MWCoin{coin});
+    return WriteIC(std::make_pair(std::string("mweb_coin"), coin.commitment), MWCoin{coin});
```
Comment (mosadialiou) [inline src/wallet/walletdb.cpp]: let's use an `enum` or at least a `constant` for `mweb_coin`

File: src/wallet/walletdb.cpp [hunk 2]
```diff
@@ -800,19 +800,31 @@ struct MWCoin {
...
 }
 
 bool WalletBatch::EraseMWCoin(const libmw::Commitment& commitment)
 {
-    return EraseIC(std::make_pair(std::string("mweb_tx"), commitment));
+    return EraseIC(std::make_pair(std::string("mweb_coin"), commitment));
```
Comment (mosadialiou) [inline src/wallet/walletdb.cpp]: let's use an `enum` or at least a `constant` for `mweb_coin`

File: src/wallet/walletdb.cpp [hunk 3]
```diff
@@ -847,7 +859,7 @@ DBErrors WalletBatch::FindMWCoins(std::vector<libmw::Coin>& coins)
 
             std::string strType;
             ssKey >> strType;
-            if (strType == "mweb_tx") {
+            if (strType == "mweb_coin") {
```
Comment (mosadialiou) [inline src/wallet/walletdb.cpp]: let's use an `enum` or at least a `constant` for `mweb_coin`

File: src/miner.cpp [hunk 4]
```diff
@@ -140,12 +140,15 @@ std::unique_ptr<CBlockTemplate> BlockAssembler::CreateNewBlock(const CScript& sc
     fIncludeWitness = IsWitnessEnabled(pindexPrev, chainparams.GetConsensus());
 
     fIncludeMWEB = IsMimblewimbleEnabled(pindexPrev, chainparams.GetConsensus());
+    if (fIncludeMWEB) {
```
Comment (mosadialiou) [inline src/miner.cpp:143]: can we write this?
```
if (fIncludeMWEB) {
       mweb_builder = libmw::miner::NewBuilder(nHeight, pcoinsTip->GetMWView());
       AddHogExTransaction(pindexPrev);
}
```

Comment (DavidBurkett) [reply src/miner.cpp:143]: No. AddHogExTransaction has to be done after addPackageTxs. And mweb_builder must be created before calling addPackageTxs.

File: src/wallet/walletdb.cpp (hunk 1 above)
Comment (DavidBurkett) [reply src/wallet/walletdb.cpp]: It actually has to be a string in order to follow the existing standard. I can definitely use a const though. even though the other tables don't. It's a better practice, and I don't mind veering a little bit from the existing code.

Comment (mosadialiou) [reply src/wallet/walletdb.cpp]: Cool!

File: src/miner.cpp (hunk 4 above)
Comment (mosadialiou) [reply src/miner.cpp:143]: I get it now

**Pull request merged**